docs/spec_draft.md

```diff
@@ -343,6 +344,120 @@ logical operation.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.batch_norm_grad
+
+### Semantics
+
+Computes gradients of `operand`, `scale` and offset, using `mean`, `variance`
+and `grad_output`, the gradient of output from `batch_norm_training`, and
+produces `grad_operand`, `grad_scale` and `grad_offset` tensors. More formally,
+this operation can be expressed as a decomposition to existing StableHLO
+operations using Python-like syntax as follows:
+
+```python
+def reduce_sum(operand, feature_index):
+  (sum,) = reduce(
+      inputs=[operand],
+      init_values=[0],
+      dimensions=[i for i in range(rank(operand)) if i != feature_index],
+      body=lambda x, y: add(x, y))
+  return sum
+
+def compute_mean(operand, feature_index):
+  sum = reduce_sum(operand, feature_index)
+  divisor = constant(num_elements(operand) / dim(operand, feature_index))
+  divisor_bcast = broadcast_in_dim(divisor, [], shape(sum))
+  return divide(sum, divisor_bcast)
+
+def batch_norm_grad(operand, scale, mean, variance, grad_output, epsilon, feature_index):
+  # Broadcast inputs to shape(operand)
+  scale_bcast = broadcast_in_dim(scale, [feature_index], shape(operand))
+  mean_bcast = broadcast_in_dim(mean, [feature_index], shape(operand))
+  variance_bcast = broadcast_in_dim(variance, [feature_index], shape(operand))
+  epsilon_bcast = broadcast_in_dim(constant(epsilon), [], shape(operand))
+
+  centered_operand = subtract(operand, mean_bcast)
+  stddev = sqrt(add(variance_bcast, epsilon_bcast))
+  normalized_operand = divide(centered_operand, stddev)
+
```

Comment:
Add comments here like the code in https://github.com/openxla/stablehlo/blob/main/docs/spec_draft.md#stablehlobatch_norm_inference

Reply:
The formula/algorithm here is not intuitive like batch_norm_training, so comments might only state the obvious as to what the code is doing.